packages/orderbook/src/orderbook.ts

```diff
@@ -160,6 +163,104 @@ export class Orderbook {
     };
   }
 
+  /**
+   * Get required transactions and messages for signing to facilitate creating bulk listings.
+   * Once the transactions are submitted and the message signed, call the completeListings method
+   * provided in the return type with the signature. This method supports up to 20 listing creations
+   * at a time. It can also be used for individual listings to simplify integration code paths.
+   * @param {PrepareBulkListingsParams} prepareBulkListingsParams - Details about the listings
+   * to be created.
+   * @return {PrepareBulkListingsResponse} PrepareListingResponse includes
+   * any unsigned approval transactions, the typed bulk order message for signing and
+   * the createListings method that can be called with the signature to create the listings.
+   */
+  async prepareBulkListings(
+    {
+      makerAddress,
+      listingParams,
+    }: PrepareBulkListingsParams,
+  ): Promise<PrepareBulkListingsResponse> {
+    // Limit bulk listing creation to 20 orders to prevent API and order evaluation spam
+    if (listingParams.length > 20) {
+      throw new Error('Bulk listing creation is limited to 20 orders');
+    }
+
+    // In the event of a single order, delegate to prepareListing as the signature is more
+    // gas efficient
+    if (listingParams.length === 1) {
+      const prepareListingResponse = await this.prepareListing({
+        buy: listingParams[0].buy,
+        makerAddress,
+        sell: listingParams[0].sell,
+        orderExpiry: listingParams[0].orderExpiry,
+      });
```

Comment:
**Suggestion:** For colocation clarity, could we consider lifting the internal call so the param manipulation matches the bulk operation below?

```suggestion
      const prepareListingResponse = this.seaport.prepareSeaportOrder(
        makerAddress,
        listingParams[0].sell,
        listingParams[0].buy,
        new Date(),
        listingParams[0].orderExpiry || new Date(Date.now() + 1000 * 60 * 60 * 24 * 365 * 2),
    );
```